### ISSUE TITLE
Getting freeze at the immersive mode in http://moonrider.xyz 

### ISSUE THREAD

**Issue by daoshengmu** [MozillaReality/FirefoxReality#2320]
## Configuration

Firefox Reality version: FxR 7
Firefox Reality build ID: 485355cc

Hardware: 
Oculus Quest

## Steps to Reproduce

1. Visit http://moonrider.xyz using the Oculus Browser on Quest
2. Click to enter VR (glasses icon in bottom right corner) and press Begin UI.
3. Go to Punch mode, and select a music to start.

## Current Behavior

Once punching a block, the screen buffer will be freeze.

## Expected Behavior

I checked it works properly in Oculus Browser, so we should be no problem with it.

## Error Logs and Stack Traces
I would guess it would be related to gamepad haptic feedback or SurfaceTexture.

```
2019-11-19 12:38:06.770 5876-5970/org.mozilla.vrbrowser E/GeckoConsole: [JavaScript Error: "TypeError: this.gamepad.hapticActuators[n.actuatorIndex] is undefined" {file: "https://moonrider.xyz/build/build.js" line: 247}]
    pulse@https://moonrider.xyz/build/build.js:247:24084
    onHit@https://moonrider.xyz/build/build.js:16:15640
    checkCollision@https://moonrider.xyz/build/build.js:16:9979
    tick@https://moonrider.xyz/build/build.js:16:9616
    value@https://moonrider.xyz/vendor/aframe-master.min.js:265:11226
    value@https://moonrider.xyz/vendor/aframe-master.min.js:265:11760
    [172]</module.exports/</<@https://moonrider.xyz/vendor/aframe-master.min.js:396:132
    t@https://moonrider.xyz/vendor/aframe-master.min.js:90:3246
```

```
```

</details>

**Comment by bluemarvin** [comment]
We don't use EGLTextures, it's most likely the unhandled error.

**Comment by bluemarvin** [comment]
We should still be able to exit immersive mode which seems to break when the page does.

**Comment by daoshengmu** [comment]
Alright, the right fixes is here, #2198. Waiting for the haptic feedback support.

**Comment by bluemarvin** [comment]
The page crashing should not block us from exiting immersive mode. We should fix that first.

**Comment by daoshengmu** [comment]
> The page crashing should not block us from exiting immersive mode. We should fix that first.

It looks odd. `const bool menuPressed = (controllerState.inputState.Buttons & ovrButton_Enter) != 0` is always false after the page crashing.

**Comment by bluemarvin** [comment]
I think we are trying to exit but something is blocking it. When I pause with the oculus home button I see the spinner for the preview.

**Comment by daoshengmu** [comment]
We are blocking by the VR immersive state [1]. It looks like OVR needs to call `vrapi_SubmitFrame2` to end this frame's state, otherwise, we can't get new button states.

[1] https://github.com/MozillaReality/FirefoxReality/blob/9727203eb2521cb854ad007eb57609bf24500373/app/src/main/cpp/BrowserWorld.cpp#L1412

**Comment by bluemarvin** [comment]
Is this Oculus only or does the same thing happen on HTC?

**Comment by daoshengmu** [comment]
> Is this Oculus only or does the same thing happen on HTC?

Unfortunately, they don't support Focus+ controllers, they didn't handle any actions from the controllers.

**Comment by daoshengmu** [comment]
Focus+ doesn't have this problem. Checked via https://mortimergoro.github.io/webvr_long_pause/.

**Comment by MortimerGoro** [comment]
Reopening until #2198 lands

**Comment by daoshengmu** [comment]
Resolved by #2198.
